controllers/argocd/route.go

```diff
@@ -366,3 +381,42 @@ func (r *ReconcileArgoCD) reconcileApplicationSetControllerWebhookRoute(cr *argo
 	}
 	return r.Client.Update(context.TODO(), route)
 }
+
```

Comment:
One final, simple request, can you add this as a comment to the top of this function:
```
The algorithm used by this function is:
- If the FIRST label ("console-openshift-console" in the above case) is longer than 63 characters, shorten (truncate the end) it to 63.
- If any other label is longer than 63 characters, return an error
- After all the labels are 63 characters or less, check the length of the overall hostname:
    - If the overall hostname is > 255, then shorten the FIRST label until the host name is < 255
    - After the FIRST label has been shortened, if it is < 20, then return an error (this is a sanity test to ensure the label is likely to be unique)
```

This will help anyone else working on this function against quickly understand what it is supposed to do.

Reply:
Added the comment @jgwest